android/source/VideoLocker/src/org/edx/mobile/model/registration/RegistrationFormField.java

```diff
@@ -0,0 +1,67 @@
+package org.edx.mobile.model.registration;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RegistrationFormField {
+    private @SerializedName("required")         boolean required;
+    private @SerializedName("name")             String name;
+    private @SerializedName("placeholder")      String placeholder;
+    private @SerializedName("defaultValue")     String defaultValue;
+    private @SerializedName("restrictions")     RegistrationRestriction restriction;
+    private @SerializedName("errorMessages")    ErrorMessage errorMessage;
+    private @SerializedName("instructions")     String instructions;
+    private @SerializedName("type")             String fieldType;
+    private @SerializedName("label")            String label;
+    private @SerializedName("options")          List<RegistrationOption> options; // may be empty
+    private @SerializedName("defaultOption")    RegistrationOption defaultOption; // may be nil
+
+    public boolean isRequired() {
+        return required;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getPlaceholder() {
+        return placeholder;
+    }
+
+    public RegistrationOption getDefaultOption() {
+        return defaultOption;
+    }
+
+    public RegistrationRestriction getRestriction() {
+        return restriction;
+    }
+
+    public ErrorMessage getErrorMessage() {
+        return errorMessage;
+    }
+
+    public String getInstructions() {
+        return instructions;
+    }
+
+    public String getFieldType() {
+        return fieldType;
```

Comment:
Let's convert to a RegistrationFieldType here so that anyone who uses this code can work with the nice enum instead of strings. It'll simplify the activity code.